torch/_meta_registrations.py

```diff
@@ -143,3 +132,86 @@ def meta_repeat_interleave_Tensor(repeats, output_size=None):
             "cannot repeat_interleave a meta tensor without output_size"
         )
     return repeats.new_empty(output_size)
+
+@torch.library.impl(meta_lib, "index.Tensor")
+def meta_index_Tensor(self, indices):
+    # aten::index is the internal advanced indexing implementation
+    check(len(indices) <= self.ndim, lambda: f"too many indices for tensor of dimension {self.ndim} (got {len(indices)})")
+    # checkIndexTensorTypes
+    result = []
+    for i, index in enumerate(indices):
+        if index is not None:
+            check(
+                index.dtype in [torch.long, torch.int8, torch.bool],
+                lambda: "tensors used as indices must be long, byte or bool tensors"
+            )
+            if index.dtype in [torch.int8, torch.bool]:
+                nonzero = index.nonzero()
+                for j in range(index.ndim):
+                    result.append(nonzero.select(1, j))
+            else:
+                result.append(index)
+        else:
+            result.append(index)
```

Comment:
Missing a here as well. This can be merged with the previous one if you add an `if` in the for loop.
```python
    check(len(results) <= self.ndim,
          lambda: f"too many indices for tensor of dimension {self.ndim}")
```

Reply:
not sure I understand the suggestion here

Reply:
technically, this check actually happens in applySlicing called from THPVariable_getitem and aten::index itself doesn't do the test and ends up with an obscure error

```
$ python -c "import torch; torch.ops.aten.index.Tensor(torch.randn(2), [torch.tensor([[True,False],[False,True]])])"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/raid/ezyang/pytorch-scratch2/torch/_ops.py", line 48, in __call__
    return self._op(*args, **kwargs or {})
IndexError: Dimension out of range (expected to be in range of [-1, 0], but got 1)
```

Reply:
So, this error might come from a few places, but perhaps the simplest example is
```python
import torch

shape = (2,)
x = torch.randn(shape)
print(x[:, :].shape)
```